### PR TITLE
Add missing advanced-enumeration files

### PR DESCRIPTION
**Pull request by angelikatyborska**
Closes https://github.com/exercism/ruby/issues/1177

On top of that issue, I also looked through all of the student-facing concept and concept exercise markdown files and tried to fix the code blocks (triple `):
- specify the language when missing
- remove syntax errors
	- replace foreign `//` comment syntax with the correct `#` :)
	- mark outputs and multiline returns as multiline comments
	- in general what is not valid Ruby code should be a comment where possible, for the sake of syntax coloring
- consistently use `# =>` (or `#=>` without a space) to signify return values
- consistently use `#` for behavior descriptions
- remove semicolons

There were also two leftover `after.md` files so I dealt with them :)

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]

File: concepts/advanced-enumeration/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "blurb": "Enumeration also works for hashes and can be nested.",
+  "authors": ["iHiD"],
```
Comment (ErikSchierboom) [inline concepts/advanced-enumeration/.meta/config.json:3]:
```suggestion
  "authors": ["angelikatyborska"],
```

**Review by ErikSchierboom — APPROVED** [review]
Thanks a ton. One tiny nit.

File: concepts/advanced-enumeration/.meta/config.json (hunk 1 above)
Comment (angelikatyborska) [reply concepts/advanced-enumeration/.meta/config.json:3]: All of the text was copied from the exercise, which was written by Jeremy 🙂 I don't want to be blamed for the content 😁

Comment (ErikSchierboom) [reply concepts/advanced-enumeration/.meta/config.json:3]: Ah, okay. Makes sense :D

**Comment by ErikSchierboom** [comment]
Once again: thanks!

**Pull request merged**